sdk/contracts/index.ts

```diff
@@ -0,0 +1,134 @@
+import { NetworkId, NetworkNameById } from '@synthetixio/contracts-interface';
+import { Contract } from 'ethers';
+
+import {
+	ExchangerABI,
+	SystemStatusABI,
+	ExchangeRatesABI,
+	SynthUtilABI,
+	SystemSettingsABI,
+	SynthRedeemerABI,
+	FuturesMarketDataABI,
+	FuturesMarketSettingsABI,
+	FuturesMarketABI,
+	SynthABI,
+} from './abis/main';
+
+export type ContractName =
+	| 'Exchanger'
+	| 'SystemStatus'
+	| 'ExchangeRates'
+	| 'SynthUtil'
+	| 'SystemSettings'
+	| 'SynthRedeemer'
+	| 'FuturesMarketData'
+	| 'FuturesMarketSettings'
+	| 'FuturesMarket'
+	| 'Synth';
+
+export type ContractDetails = {
+	name: ContractName;
+	networks: Partial<Record<NetworkId, { address: string; contract: Contract }>>;
+};
+
+export type AllContractsMap = Record<ContractName, Partial<Record<NetworkId, Contract>>>;
+
+export type ContractMap = Partial<Record<ContractName, Contract>>;
+
+export const contracts: AllContractsMap = {
+	Exchanger: {
+		1: new Contract('0xD64D83829D92B5bdA881f6f61A4e4E27Fc185387', ExchangerABI),
```

Comment:
That's interesting. In the future, we will still need to copy the addresses into a file somehow, so it might be better to just have them all in one place. We can definitely talk more about this when we start refining this, though.